union/core/core.py

```diff
@@ -0,0 +1,126 @@
+import inspect
+import io
+import sys
+import traceback
+
+import aiohttp
+import discord
+from discord.ext import commands
+
+
+class Core:
+
+    """
+    Core features of the bot.
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.session = aiohttp.ClientSession(loop=bot.loop)
```

Comment:
Basically doing it per-class means that it takes up slightly more ram, plus a bit of time to load the cog. Possible security issues with doing it botwide, not 100% clear on that, but danny brought up token leaking as a reason not to use bot.http.session, which is separate from our session anyway.